### PR TITLE
Add assertions before potential null deferences

### PR DESCRIPTION
**Pull request by meshcollider**
Picked up by the static analyzer [Facebook Infer](http://fbinfer.com/) which I was playing around with for another research project. Just adding some asserts before dereferencing potentially null pointers.

### CODE REVIEW

**Comment by JeremyRubin** [comment]
Seems reasonable, utAck.

File: src/addrdb.cpp [hunk 1]
```diff
@@ -49,7 +49,10 @@ bool SerializeFileDB(const std::string& prefix, const fs::path& path, const Data
         return error("%s: Failed to open file %s", __func__, pathTmp.string());
 
     // Serialize
-    if (!SerializeDB(fileout, data)) return false;
+    if (!SerializeDB(fileout, data)) {
+        fileout.fclose();
```
Comment (laanwj) [inline src/addrdb.cpp]: This is an autofile - doesn't it close automatically on release?

Comment (promag) [reply src/addrdb.cpp]: Maybe remove explicit close and move `fileout` to an inner scope?

**Comment by practicalswift** [comment]
utACK modulo the autofile fix which is redundant.

Generally I think it is a good very idea to add `assert(…)`:s where it help static analyzers understand the code better. It makes implicit assumptions explicit which helps static analyzers and humans alike to understand and reason about the code.

My experience is that if a code path makes a static analyzer confused it is very likely to be confusing for humans too. This holds even in a lot of cases where it could be argued that the static analyzer should be "smarter" and in some way misunderstands the code. Confusion leads to bugs, so better avoid confusion by liberal use of `assert`:s and other mechanisms to make tacit assumptions explicit.

**Comment by MeshCollider** [comment]
Removed redundant file close as requested, thanks for the reviews :)

**Comment by practicalswift** [comment]
utACK c0019924406e1ce8368465c768de11019ad5eeed

**Comment by promag** [comment]
utACK c001992.

**Comment by laanwj** [comment]
Can you please change the title to `Add nullptr assertions` or such? The current title is incorrect, as adding assertions does not *fix* anything.

**Comment by laanwj** [comment]
utACK c00199244

**Pull request merged**